Fix link, css, and validator errors (prep for publication)

aphillips opened the pull request:
- fixed use of `indent` to `text-indent`
- moved local-to-HTML styling to local.css and replaced with a class
- fixed typo of `jang` => `lang`


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/bp-i18n-specdev/pull/92.html" title="Last updated on Apr 21, 2023, 9:42 AM UTC (b6b3fc5)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/bp-i18n-specdev/92/de5bee9...aphillips:b6b3fc5.html" title="Last updated on Apr 21, 2023, 9:42 AM UTC (b6b3fc5)">Diff</a>

aphillips committed:
Fix link, css, and validator errors (prep for publication)

- fixed use of `indent` to `text-indent`
- moved local-to-HTML styling to local.css and replaced with a class
- fixed typo of `jang` => `lang`

aphillips committed:
Address @r12a's comments on #89
- Fix style to relative size
- Reword guidance to implementers
- Reword about sorting numbers and dates to say "internal
  representation"
- Added a subsection for "searching" and put up the "under construction"
  banner

aphillips committed:
Fix typo `CVS` => `CSV`

aphillips committed:
First pass at 2023-03-30 changes

- Move the spillover example
- Move the section on strcat to "Text Processing" and add a "see also"
- Fix terminology markup in 3.2.3
- Fix the `Suppress-Script` typo
- Add `FSI` to the last para of 3.2.3

This is not the finished product. I need to do something smarter with
the spillover example and I need to add the third bit of mustard we
discussed in the call.

aphillips committed:
Step II: fix links, exchange markup/controls

xfq commented on index.html:
```diff
@@ -2772,11 +2772,11 @@ <h3>Concatenation of strings</h3>
 </aside>
 
-<p>See also: [:bidi_inline:]</p>
+<p>See also: &#xa7;[[[#inline_changes]]] and &#xa7;[[[#bidi_inline]]]</p>
```
If it's better to add the section sign (§), we should probably file an issue in ReSpec and let ReSpec add it automatically in the future.

r12a replied:
We don't normally add the section sign, and i don't see any reason to add it.

Why are we using `&#xa7;` rather than § in the source code, anyway?

aphillips replied:
In theory, ReSpec is supposed to add it, but it appears to be borken. I'll file an issue on them and remove here.

This needs to be moved to the see also box anyway.

r12a replied:
Just realised that i forgot to add `..` to my question about §.  I edited it. It might be easier to understand what i was asking now.

r12a replied:
And btw, fwiw i was never actually a fan of the use of § by respec.

aphillips replied:
I assumed that you were referring to my use of the entity, but yeah, that makes the comment clearer. I used an entity because I tend to non-ASCII characters in source code. In any case, removing the silcrows.

r12a commented on index.html:
```diff
@@ -903,11 +928,13 @@ <h3>Setting base direction for inline or substring text</h3>
 
  	<div class="req" id="bidi_inline_auto">
-	<p class="advisement">It must be possible to also set the direction for a span to auto. This means that the base direction will be determined by examining the content itself. A typical approach here would be to set the direction based on the first strong directional character outside of any markup.</p>
+	<p class="advisement">It must be possible to also set the direction for a span to <span class="kw" translate="no">auto</span>. This means that the base direction will be determined by examining the content itself. A typical approach here would be to set the direction based on the first strong directional character outside of any markup.</p>
```
span probably needs to be .kw too

aphillips replied:
@r12a not sure I understand. It's already class `kw`. I used `span` instead of `code` or `kbd` because it's not actually the keyword here--it's being used somewhat casually as a synonym for first-strong. I think probably I'll revert this line for now.

r12a replied:
You put .kw around `auto`, but not around the word `span` (just before).

aphillips replied:
Ahh!!! Although I think the word span there doesn't mean `<span>` but just "a range of text"?

aphillips committed:
Remove silcrows and style links as see also

aphillips committed:
Mild rewording to address the span/auto text.

aphillips committed:
Fixing the spillover/interference example
- Remove part of the example string
- Use Arabic commas (U+060C) in the Arabic
- Remove some of the text and rewrite other bits to make generic
- Move the old external-link examples to see also
- Style the control character names, use the elipsis character
- Minor tweaking to make clear that RLI/LRI/FSI are isolating and RLE
  etc are embedding

aphillips committed:
Clean up code point example table

aphillips committed:
Updating the character template
- remove all square brackets
- fix a couple of places where we failed to use the template ourselves
- update the styles in local.css for testing purposes
- clean various uses of `code` etc. where we should use `bdi` or `span`

aphillips committed:
Add instructions for invisibles and sequences.

aphillips committed:
Fix character template in the "family" examples.

r12a commented:
lines 2194, 2212: add class="kw" to the code element
line 2216: we should probably add that this is a special case, and it is normally more useful to include the character name(s).  In fact, we may want to advise people not to use just code point values, but to use characters and names as well as a general good practise.

Btw, this PR is getting so big that i'm beginning to forget what i have reviewed and what i haven't.  I think multiple PRs would be better in future, if possible.

r12a commented:
Also, there are places where the codepoint markup children appear to have a space between them, and others where they don't.  Since you have added a margin to the style sheet, we should probably consistently remove the intervening space (in code examples and in descriptive examples).

There's also the question of what to do for combining marks.  These can often look a mess if you try to use them as characters, and may need one or sometimes several &amp;nbsp's to make them look right.  I have found that a safer option is to use an svg or png image, instead.

aphillips commented:
@r12a I made changes on lines 2194 and 2212 that you suggest. I reworded and clarified on 2216 along the lines of what you're suggesting (have a look).

I agree about the problem with spaces and I'm open to suggestions. I added the half-`em` of space in the `margin-inline` to guard against problems when the space is accidentally omitted. When the space is present one gets something more like 1.5 em of space. I didn't want to fool with trying to collapse the space. Overall this is a side-effect of removing the square brackets. Should we just say "put a space before the span" and remove the `margin-inline` space?

Regarding combining marks, I agree that's a problem. Should I add text suggesting the need to include a no-break space or other base character (such as how you included a base as an example with SARA AM in the glossary recently)?

aphillips committed:
Address @r12a's comments on the character styles

aphillips committed:
One more tweak to the instructions for character formatting

r12a commented on index.html:
```diff
@@ -2209,14 +2209,25 @@ <h5>Character naming template</h5>
+
+</aside>
 
+<p>Long sequences will sometimes just list the code points, although the character names should be retained where possible for clarity. An example can be found in this doucment in the <a href="#char_term_def">discussion of the composed "family" emoji</a>: <span class="codepoint"><bdi lang="en">👨‍👩‍👧‍👧</bdi> <span class="uname">U+1F468 U+200D U+1F469 U+200D U+1F467 U+200D U+1F467</span></span></p>
```
doucment -> document

r12a replied:
For the spacing, i think we need to either (a) say add a space and remove the margin styling, or (b) say use no space and keep the margin spacing. I don't think we should have both spaces and margins.  After observing to myself that authors may automatically put spaces, so A may be better, i thought some more and decided that B gives more control over spacing. For example, we'd be able to increase or decrease the size of the gap in future, or on an element-by-element basis by adjusting the styling. Adjusting the margin on the element tag provides a much better solution, i think, for dealing with combining characters than using `&nbsp;`, ie. by putting margin before the `bdi` tag, although sometimes it is needed after.

Another way to handle the combining marks is to suggest that author copy an image from my Unicode character image set and use that.  Note that this can also be used for most non-space invisibles, too.  We'd need to provide some markup for that which replaces the bdi with an img element.  The styling for that would need to remove both spaces and margins, since those images are all square.

Sorry that i'd forgotten how much thought i had to put into this stuff to come up with what i currently do.  And the margins replacing nbsp is a new thought, and i suspect a good one.

aphillips replied:
Having slept on this... I think the thing to do here is to move the margins to the `bdi` element. People will naturally include a space before/after the uname span when omitting the exemplar character (just as they will put a space before/after the codepoint span all of the time). Instead of stage managing the space before the uname, we can manage the space around the example character instead.

Check out the latest push on this PR. I've removed all of the spaces between `bdi` and `span` and styled the `bdi` with 0.5em after margin (recall that the font is at 130%). Standalone uses of `uname` in the doc look normal. We might tweak the size/space a bit on the `bdi`? Thoughts?

r12a replied:
Moving the margin to the bdi is certainly the right thing to do - sorry i forgot to mention that earlier.

However, if the bdi contains RTL text the different amounts of spacing won't appear in the expected location, relative to the uname.  So i think you need margin-inline:.5em;

For my own documents, I'm also thinking about extensions which allow you to add a class name which will add extra space to the left or right of the exemplar character.  This can be useful when dealing with combining marks - some of which may overlap the uname or the preceding text.  I also remove the spacing altogether when including an image, rather than a character, since the images i'm using usually have space built in (because they are square).

aphillips committed:
Fix typo

aphillips committed:
Update uname/codepoint styles based on @r12a conversation

aphillips committed:
Changing the margin-inline to be more attractive

aphillips committed:
Changes from telecon of 2023-04-13
- made 'teal' the color
- formatting of the examples so no spaces between bdi and span
- mention the need for no spaces

xfq commented on index.html:
```diff
@@ -2202,26 +2202,27 @@ <h5>Character naming template</h5>
-   &lt;span class="uname">U+064E ARABIC FATHA&lt;/span>
+&lt;span class="codepoint" translate="no">
+  &lt;bdi lang="hi">&amp;#x0928;&amp;#x093f;&lt;/bdi>&lt;span class="uname">U+0928 DEVANAGARI LETTER NA&lt;/span> + 
+  &lt;span class="uname">U+093F  DEVANAGARI VOWEL SIGN I&lt;/span>
```
```suggestion
  &lt;span class="uname">U+093F DEVANAGARI VOWEL SIGN I&lt;/span>
```

aphillips committed:
Update index.html

Co-authored-by: Fuqiao Xue <xfq@w3.org>

aphillips committed:
Suggested changes from 2023-04-19 email

aphillips committed:
Merge branch 'gh-pages' of https://github.com/aphillips/bp-i18n-specdev into gh-pages

r12a commented on index.html:
```diff
@@ -2800,7 +2800,7 @@ <h3>Concatenation of strings</h3>
 	<p>Here is an example of a Note that you can include in your specification when recommending that implementations generate displayable [=natural language=] values:</p>
 	<div class="note" role="note" id="bidi-gen-note-example">
-	   <p class="example_note">If values generated for <em><code>_field_name_</code></em> contain or might contain [= bidirectional text =], [=producers=] should ensure that [=consumers=] of the string will display it correctly without having to process or alter the string. This can be done using markup or via the insertion of Unicode bidirectional controls. In addition, supplying the correct base d ...
+	   <p class="example_note">When generating a value for <em><code>_field_name_</code></em>, [=producers=] need to ensure that any [=bidirectional text=] inside the string will display correctly. When the value is assembled from multiple strings, individual substrings might need to be bidi isolated in order to prevent [=spillover=] effects. This can be done using markup (where available) or via th ...
```
I'd be inclined to emphasize the word 'inside', just to make it clear.  We could also talk about changes of direction, rather than bidi text, but that's just a thought.

r12a replied:
> When the value is assembled from multiple strings, individual substrings might need to be bidi isolated in order to prevent [=spillover=] effects.

Do you mean here by the producer or by the consumer?  It's not clear to me.

Think a bit further, i wonder whether introducing something like the producer/consumer bullet points made for SPC would help here?

r12a replied:
> In addition, supplying the correct base direction as metadata allows the [=consumer=] to both isolate the string and set its direction, avoiding [=spillover=] effects between the full string and the context in which it is eventually displayed.

Here it might help to say something like: In addition, if the producer can provide metadata that indicates the expected base direction for the string as a whole, the consumer will be able to both isolate ....

aphillips replied:
Excellent comments. I think I addressed them in the latest push.

aphillips committed:
Address @r12a's comments

aphillips committed:
Address comments about #strcat from 2023-04-20 call

r12a commented on index.html:
```diff
@@ -2800,7 +2800,7 @@ <h3>Concatenation of strings</h3>
 	<p>Here is an example of a Note that you can include in your specification when recommending that implementations generate displayable [=natural language=] values:</p>
 	<div class="note" role="note" id="bidi-gen-note-example">
-	   <p class="example_note">When generating a value for <em><code>_field_name_</code></em>, [=producers=] need to ensure that any [=bidirectional text=] <em>inside</em> the string will display correctly. When a [=producer=] assembles the value from multiple strings, individual substrings might need to be [=bidi isolated=] in order to prevent [=spillover=] effects. This can be done using markup (w ...
+	   <p class="example_note">When generating a value for <em><code>_field_name_</code></em>, [=producers=] need to ensure that directional changes <em>inside</em> the string will display correctly. When a [=producer=] assembles the value from multiple strings, individual substrings might need to be [=bidi isolated=] in order to prevent [=spillover=] effects. This can be done using markup (where av ...
```
Inlines -> inline

aphillips committed:
Address character styles from 2023-04-20 call
- use Arial+monospaced
- set line-height to 1 em

aphillips commented:
Merging per call of 2023-04-30. Errors/changes should be filed as new issues.
